### PR TITLE
New login auth

### PR DESCRIPTION
**Pull request by BAANG**
- Created sign in widget that uses Google for the Social Login authentication.

- Converted Victoire's routing, consolidating to a more modularized method taking advantage of the sequelize package.

### CODE REVIEW

File: routes/api-routes.js [hunk 1]
```diff
@@ -26,62 +23,62 @@ module.exports = function(app) {
...
-    }).catch(function (err) {
-        console.error(err.original.sqlMessage)
-        res.sendStatus(400)
     });
-});
 
-module.exports = router;
+    app.delete("/favorites/:id", function (req, res) {
+        console.log(req.method, `Deleting favorite for animal id ${req.params.id}`);
+        app.Favorites.destroy({
```
Comment (Victoire44) [inline routes/api-routes.js]: Why we don't use db.Favorites here?

File: routes/api-routes.js [hunk 2]
```diff
@@ -26,62 +23,62 @@ module.exports = function(app) {
...
+            res.sendStatus(200)
+        }).catch(function (err) {
+            console.error(err.original.sqlMessage)
+            res.sendStatus(400)
+        });;
+    });
+
+    app.get("/favorites/:id", function (req, res) {
+        console.log(req.method, `Getting favorite for animal id ${req.params.id}`);
+        app.Favorite.findOne({
```
Comment (Victoire44) [inline routes/api-routes.js]: Why we don't use db.Favorites here?

File: routes/api-routes.js [hunk 3]
```diff
@@ -26,62 +23,62 @@ module.exports = function(app) {
...
-});
 
-router.get("/favorites/:id", function (req, res) {
-    console.log(req.method, `Getting favorite for animal id ${req.params.id}`);
-    Favorite.findOne({
-        where: {
+    // =============================================================
+    app.post("/favorites/:id", function (req, res) {
+        console.log(req.method, `Adding favorite for animal id ${req.params.id}`);
+        db.Favorites.create({
```
Comment (Victoire44) [inline routes/api-routes.js]: Why we don't use `db.Favorite` without a `s` here?

File: models/favorite.js [hunk 4]
```diff
@@ -1,17 +1,8 @@
-// Dependencies
-// =============================================================
-var Sequelize = require("sequelize");
-var db = require("../config/connection.js");
-var User = require("../models/users.js");
+module.exports = function (sequelize, DataTypes) {
+    var Favorite = sequelize.define("Favorites", {
+        animalId: DataTypes.STRING
+    });
+    // Favorite.belongsTo(Users)
```
Comment (Victoire44) [inline models/favorite.js]: We should probably keep this as it adds the `userId` to the table.
To test I created a user with this query:
```
INSERT into purrfect_match_DB.users (username, password, name, location, favorites, preferences, has_preferences) 
values ("toto", "user-id", "toto", "San Francisco", null, null, false);
```

**Comment by BAANG** [comment]
Resolved the changes you mentioned above, minus the Favorite.belongsTo() function. Node was spitting back an error when I enabled that code. So we'll work on that bit soon.

**Comment by BAANG** [comment]
Also, @Victoire44 , the reason I removed that I changed the routes to db.Favorites is because sequelize looks for the name of the database in sql. If you'd like we can amend the database to be 'Favorite' instead of 'Favorites' but at that point it's just a matter of syntax.

File: models/users.js [hunk 5]
```diff
@@ -1,22 +1,20 @@
...
-    password: Sequelize.STRING,
-    name: Sequelize.STRING,
-    location: Sequelize.STRING,
-    favorites: Sequelize.STRING,
-    preferences: Sequelize.STRING,
-    has_preferences: Sequelize.BOOLEAN
-});
+    // Creates a "Users" model that matches up with DB
+    var Users = sequelize.define("Users", {
+        userId: DataTypes.STRING,
```
Comment (Victoire44) [inline models/users.js]: Why did you add `userId`? I think sequelize creates the field `id` for us already.
